races_bp.py

```diff
@@ -28,5 +28,5 @@ def show_races():
     if id_race:
         carpool_offer = db_functions.get_carpool_offers_for_race(id_race)
 
-    return render_template('races.html', zavody=races, datum_zavodu=race_date, id_vybraneho=id_race,
-                           nabidka_spolujizdy=carpool_offer)
+    return render_template('races.html', races=races, date_race=race_date, id_chosen=id_race,
```

Comment:
It would make sense to unify the wording, i.e. pick one of `date_race` and `race_date` and use it consistently everywhere. Since you already use `race_date` in Python, I would suggest using `race_date` in the templates as well.